### PR TITLE
networkd: route replace parse prefix with generic in_addr_prefix_from_string

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

**Comment by keszybz** [comment]
LGTM. I pushed a small tweak on top.

**Comment by poettering** [comment]
lgtm

**Comment by keszybz** [comment]
@ssahani's commit passed, and my commit only changes the string message, so we can assume that it'll pass too.

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -772,45 +772,20 @@ int config_parse_destination(const char *unit,
         if (r < 0)
                 return r;
 
-        /* Destination|Source=address/prefixlen */
-
-        /* address */
-        e = strchr(rvalue, '/');
```
Comment (evverx) [inline src/network/networkd-route.c:778]: `f` and `e` were removed and that causes the following warning:
```
src/network/networkd-route.c: In function ‘config_parse_destination’:
src/network/networkd-route.c:763:16: warning: unused variable ‘f’ [-Wunused-variable]
         int r, f;
                ^
src/network/networkd-route.c:760:31: warning: unused variable ‘e’ [-Wunused-variable]
         const char *address, *e;
                               ^
src/network/networkd-route.c:760:21: warning: unused variable ‘address’ [-Wunused-variable]
         const char *address, *e;
                     ^~~~~~~
```

Comment (keszybz) [reply src/network/networkd-route.c:778]: @ssahani, please compile your PR and squash any warnings before submitting...

**Pull request merged**